angular/src/lib/tooltip/tooltip.directive.ts

```diff
@@ -211,33 +217,34 @@ export class TooltipDirective implements OnDestroy {
       }
       this.tooltipRef.instance.direction = this.direction;
       this.tooltipRef.instance.allowHover = this.allowHover;
-      this.tooltipRef.instance.mouseLeaveEvent.subscribe(() => {
+      this.mouseLeaveEventToTooltip = this.tooltipRef.instance.mouseLeaveEvent.subscribe(() => {
         this.keepTooltipOpen = false;
         this.delay = 500;
         this.close();
-        this.tooltipRef.instance.mouseLeaveEvent.unsubscribe();
       });
-      this.tooltipRef.instance.mouseEnterEvent.subscribe(keepOpen => {
+      this.mouseEnterEventToTooltip = this.tooltipRef.instance.mouseEnterEvent.subscribe(keepOpen => {
         if ( keepOpen && this.allowHover ) {
           this.keepTooltipOpen = true;
         }
-        this.tooltipRef.instance.mouseEnterEvent.unsubscribe();
       });
+     
     }
   }
 
 
   public close() {
-      if ( !this.keepTooltipOpen ) {
-        setTimeout(() => {
-          if (this.overlayRef) {
-            this.overlayRef.detach();
-          }
-          if (this.tooltipRef ) {
-            this.tooltipRef.destroy();
-            this.tooltipRef = null;
-          }
-        }, this.delay);
+    setTimeout(() => {
+      if ( !this.keepTooltipOpen && !this.isOnTarget) {
+        if (this.overlayRef) {
+          this.overlayRef.detach();
+        }
+        if (this.tooltipRef ) {
+          this.mouseEnterEventToTooltip.unsubscribe();
```

Comment:
[suggestion] When needing to manually manage multiple subscriptions, could create your own subscription and add to it so that you only have one to unsubscribe.
```ts
private subs = new Subscription();
...
this.subs.add(/* new sub */);
this.subs.add(/* new sub */);
...
this.subs.unsubscribe();
```